src/coreclr/nativeaot/Runtime/ThunksMapping.cpp

```diff
@@ -100,13 +100,18 @@ FCIMPL0(int, RhpGetThunkBlockSize)
 }
 FCIMPLEND
 
-EXTERN_C void* QCALLTYPE RhAllocateThunksMapping()
+EXTERN_C void* QCALLTYPE RhAllocateThunksMapping(int* isOOM)
```

Comment:
It is unusual to return error code via out argument, and only allow that error code to be OOM.

Can we return proper error code, and return the pointer via `[out]` argument instead?

The error code can be an HRESULT or an enum specific to this call. (I know HRESULTs are Windows-specific, but they are part of BCL Exception so they are not going away x-plat, so there is not much value in trying hard to avoid them.)

Reply:
@AaronRobinsonMSFT deleted the HRESULT definitions in #115858 so this no longer compiles.

I would prefer going back to the bool if we don't want HRESULTs anymore.